presto-main/src/main/java/com/facebook/presto/type/LikeFunctions.java

```diff
@@ -36,6 +38,8 @@
 
 public final class LikeFunctions
 {
+    public static final String INVALID_ESCAPE_MESSAGE = "Escape character must be followed by '%%', '_' or the escape character itself";
```

Comment:
I didn't realize `checkCondition` takes a format string. Having this constant be a format string (the `%%`) is non-obvious. Let's move `isLikePattern` and `unescapeValidLikePattern ` to this class, so all of the logic is in the same place, and add a helper method:
```java
private static void checkEscape(boolean condition)
{
    checkCondition(condition, INVALID_FUNCTION_ARGUMENT, "Escape character must be followed by '%%', '_' or the escape character itself");
}